### PR TITLE
Add Purchasing Logic!

### PR DESCRIPTION
**Pull request by Jeremy-Barnes**
I think this is done and ready to go!

### CODE REVIEW

File: CritterServer/DataAccess/Db/UserDbRepository.cs [hunk 1]
```diff
@@ -79,14 +79,14 @@ public async Task<bool> UserExistsByUserNameOrEmail(string userName, string emai
...
 
-        public async Task UpdateUserCash(int userId, int deltaCash)
+        public async Task<bool> UpdateUserCash(int userId, int deltaCash)
         {
-            await dbConnection.ExecuteAsync($"UPDATE users SET cash = cash + @cash WHERE id = @userId", new { userId, cash = deltaCash });
+            return (await dbConnection.ExecuteAsync($"UPDATE users SET cash = cash + @cash WHERE id = @userId", new { userId, cash = deltaCash })) == 1;
         }
 
-        public async Task UpdateUsersCash((int UserId, int CashDelta)[] userIdAndDeltaCashAmounts)
+        public async Task<bool> UpdateUsersCash((int UserId, int CashDelta)[] userIdAndDeltaCashAmounts)
```
Comment (jillkgraves) [inline CritterServer/DataAccess/Db/UserDbRepository.cs:87]: rename userIdAndDeltaCashAmounts to cashUpdates

File: CritterServer/Domains/ShopDomain.cs [hunk 2]
```diff
@@ -91,6 +93,32 @@ public async Task RemoveItemFromShop(ShopItem item, User activeUser)
             await updateShopItem(item, activeUser, true);
         }
 
+        public async Task PurchaseShopItem(int itemId, User activeUser)
+        {
+            using (var trans = TransactionScopeFactory.Create())
+            {
+                var shopItemAndItem = (await ShopRepo.RetrieveShopItemByItemId(itemId)).FirstOrDefault();
```
Comment (jillkgraves) [inline CritterServer/Domains/ShopDomain.cs:100]: shopItemAndItem is a weird name. It looks like this is the item to be purchased, so maybe itemToPurchase ?

Comment (jillkgraves) [reply CritterServer/Domains/ShopDomain.cs:100]: Use the type instead of var

File: CritterServer/Domains/UserDomain.cs [hunk 3]
```diff
@@ -143,7 +148,12 @@ public async Task ChangeUsersCash(List<(int UserId, int CashDelta)> userIdAndCas
         {
             using (var trans = TransactionScopeFactory.Create())
             {
-                await UserRepo.UpdateUsersCash(userIdAndCashDeltas.ToArray());
+                if(!await UserRepo.UpdateUsersCash(userIdAndCashDeltas.ToArray()))
```
Comment (jillkgraves) [inline CritterServer/Domains/UserDomain.cs:151]: userIdAndCashDeltas rename to cashUpdate

**Review by jillkgraves — APPROVED** [review]

File: CritterServer/Domains/ShopDomain.cs (hunk 2 above)
Comment (Jeremy-Barnes) [reply CritterServer/Domains/ShopDomain.cs:100]: itemToPurchase => purchaseItem

**Pull request merged**